pkg/odo/cli/delete/component.go

```diff
@@ -22,15 +40,20 @@ func NewComponentOptions() *ComponentOptions {
 }
 
 func (o *ComponentOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
 }
 
 func (o *ComponentOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
-	return nil
+	createParameters := genericclioptions.NewCreateParameters(cmdline)
+	if o.name == "" {
+		createParameters = createParameters.NeedDevfile("")
+	}
+	o.Context, err = genericclioptions.New(createParameters)
```

Comment:
Could you create the context only when name is ""?
AS a metter of fact, it is not possible to use `genericclioptions.New` outside an odo directory, as it will fail due to the missing env file, and the context is not necessary anymore in this case ([see https://github.com/redhat-developer/odo/pull/5511/files](https://github.com/redhat-developer/odo/pull/5511/files#diff-1a997d7c41a88e0f0001418a841d488fe47beffdbfc7e2616dc52925568e5913R42).